libraries/load-themed-styles/src/index.ts

```diff
@@ -228,24 +257,42 @@ export function loadTheme(theme: ITheme | undefined): void {
 
 /**
  * Clear already registered style elements and style records in theme_State object
+ * @option: specify which group of registered styles should be cleared.
+ * Default to be both themable and non-themable styles will be cleared
  */
-export function clearStyles(): void {
-  _themeState.registeredStyles.forEach((styleRecord: IStyleRecord) => {
+export function clearStyles(option?: ClearStyleOptions): void {
+  option = option || ClearStyleOptions.all; // default is to clear all registered styles
+  if (option === ClearStyleOptions.all || option === ClearStyleOptions.onlyNonThemable) {
+    clearStylesInternal(_themeState.registeredStyles);
+    _themeState.registeredStyles = [];
+  }
+  if (option === ClearStyleOptions.all || option === ClearStyleOptions.onlyThemable) {
+    clearStylesInternal(_themeState.registeredThemableStyles);
+    _themeState.registeredThemableStyles = [];
+  }
+}
+
+function clearStylesInternal(records: IStyleRecord[]): void {
+  records.forEach((styleRecord: IStyleRecord) => {
     const styleElement: HTMLStyleElement = styleRecord && styleRecord.styleElement as HTMLStyleElement;
     if (styleElement && styleElement.parentElement) {
       styleElement.parentElement.removeChild(styleElement);
     }
   });
-  _themeState.registeredStyles = [];
 }
 
 /**
  * Reloads styles.
  */
 function reloadStyles(): void {
   if (_themeState.theme) {
-    for (const styleRecord of _themeState.registeredStyles) {
-      applyThemableStyles(styleRecord.themableStyle, styleRecord);
+    const themableStyles: ThemableArray[] = [];
+    for (const styleRecord of _themeState.registeredThemableStyles) {
+      themableStyles.push(styleRecord.themableStyle);
+    }
+    if (themableStyles.length > 0) {
+      clearStyles(ClearStyleOptions.onlyThemable);
+      applyThemableStyles([].concat.apply([], themableStyles));
```

Comment:
this is the key part of improvement. Before this change, we replace 100 of these style elements which is slow in IE
now, we delete them which is fast and replace them with only 1 style element after theming